### PR TITLE
i18n: Markup CorrelationsPage for translations

### LINKED ISSUE #75444: i18n: Mark up CorrelationsPage for translations
### Requirements ###

Add support for i18n for CorrelationsPage and CorrelationsFeatureToggle component. All text presented in the UI should have i18n keys.

More details in #71535

### PR DESCRIPTION
**Pull request by khushijain21**
**What is this feature?**

i18n translations

**Who is this feature for?**

Fixes #75444

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/correlations/CorrelationsPage.tsx [hunk 1]
```diff
@@ -157,7 +158,9 @@ export default function CorrelationsPage() {
       navModel={navModel}
       subTitle={
         <>
-          Define how data living in different data sources relates to each other. Read more in the{' '}
+          <Trans i18nKey="correlations.sub-title">
```
Comment (ifrost) [inline public/app/features/correlations/CorrelationsPage.tsx:161]: Links should probably be part of the translations to allow linking to docs with a specific locale. Also, in this case the link label is not translated:

Screenshot 2023-09-19 at 12 17 14

File: public/app/features/correlations/CorrelationsPage.tsx [hunk 2]
```diff
@@ -279,7 +289,7 @@ const InfoCell = memo(
     const readOnly = props.row.original.provisioned;
 
     if (readOnly) {
-      return <Badge text="Read only" color="purple" className={noWrap} />;
+      return <Badge text={t('correlations.badge.text', 'Read only')} color="purple" className={noWrap} />;
```
Comment (ifrost) [inline public/app/features/correlations/CorrelationsPage.tsx:292]: In this case the key should probably indicate the message on the badge. I think something like `correlations.list.read-only` would be more meaningful. We could also use `correlations.list.` to translate other labels in the list (deleting correlations, column names, etc.)

**Review by ifrost — COMMENTED** [review]
Thanks a lot @khushijain21 for contributing! I left a few comments about key names and links in translations. I also changed the description of the PR (to mark it as related PR, not fully solving the issue). It won't solve the main issue yet. The full scope requires translating forms as well, but it could be added in a separate PR.

File: public/app/features/correlations/CorrelationsPage.tsx [hunk 3]
```diff
@@ -157,9 +158,12 @@ export default function CorrelationsPage() {
       navModel={navModel}
       subTitle={
         <>
-          Define how data living in different data sources relates to each other. Read more in the{' '}
+          <Trans i18nKey="correlations.sub-title">
+            Define how data living in different data sources relates to each other. Read more in the
```
Comment (ifrost) [inline public/app/features/correlations/CorrelationsPage.tsx:162]: Links to docs should be part of the translation. You can see an example in how `time-picker.content.empty-recent-list-docs` was introduced.

**Review by L-M-K-B — COMMENTED** [review]
Looks great from Frontend Platform POV. 🚀

File: public/app/features/correlations/CorrelationsPage.tsx [hunk 4]
```diff
@@ -157,10 +158,17 @@ export default function CorrelationsPage() {
       navModel={navModel}
       subTitle={
         <>
-          Define how data living in different data sources relates to each other. Read more in the{' '}
-          <a href="https://grafana.com/docs/grafana/next/administration/correlations/" target="_blank" rel="noreferrer">
-            documentation <Icon name="external-link-alt" />
-          </a>
+          <Trans i18nKey="correlations.sub-title">
+            Define how data living in different data sources relates to each other. Read more in the
```
Comment (ifrost) [inline public/app/features/correlations/CorrelationsPage.tsx:162]: An extra space will be needed before the link (not after). And translations should be re-generated, at the moment JSON contains:

```
    "sub-title": "Define how data living in different data sources relates to each other. Read more in the"
```

and the link to docs is not visible

Screenshot 2023-09-28 at 13 06 32

Comment (khushijain21) [reply public/app/features/correlations/CorrelationsPage.tsx:162]: hey, I ran yarn i18n:extract but the JSON still container the same value. Do you know why it is not generating "documentation"

Comment (ifrost) [reply public/app/features/correlations/CorrelationsPage.tsx:162]: Hmm... `i18n:extract` seems to be adding only new translations and does nothing when it already exists. Quite annoying during development of new translations :D I guess in such case, when it's the first time translation is added and only English translations are added you could probably revert all changes in grafana.json files (or remove the keys manually) and re-rerun `i18n:extract` - @grafana/grafana-frontend-platform is there a better way to do it?

Comment (khushijain21) [reply public/app/features/correlations/CorrelationsPage.tsx]: okay, just did it. lmk

File: public/app/features/correlations/CorrelationsPage.tsx (hunk 1 above)
Comment (joshhunt) [reply public/app/features/correlations/CorrelationsPage.tsx:161]: (FYI, we don't support this. There's no need to translate docs links)

**Review by ifrost — APPROVED** [review]
Great work! 🚀

**Pull request merged**